### ISSUE TITLE
Memoize does not work for bigint

### ISSUE THREAD

**Issue by kessido** [JuliaCollections/Memoize.jl#74]
I wanted to let you know that @memoize for function that recive bigint does not act as expected.

![image](https://user-images.githubusercontent.com/16467000/106674631-4a50d300-65bc-11eb-91cb-292b007a063d.png)

**Comment by cstjean** [comment]
Yes, I suppose that is a consequence of the `IdDict` default. Does it work if you `@memoize Dict() function f(x) ...`?

**Comment by kessido** [comment]
Yes, Tho I would suggest using Dict by default then as it add some underling logic prone to bugs.

![image](https://user-images.githubusercontent.com/16467000/106948381-4e076580-6734-11eb-9f1a-5389588cddfa.png)

**Comment by cstjean** [comment]
Yeah... I didn't make that call. It would be a breaking change, but I think I agree with you there. `IdDict` is IMO premature optimization for this package.
